src/components/Garden.js

```diff
@@ -1,136 +1,162 @@
+    const petalPathOne = 'M0,-100 C50,-60 50,-30 20,0 L0,-15 L-20,0 M0,-100 C-50,-60 -50,-30 -20,0';
+    const petalPaths = [petalPathOne, petalPathTwo, petalPathThree];
+    const petalRotationStarts = [0,60,0];
+
+    const flowerPositionBox = flowerBed.selectAll('.flower-box')
+      .data(repositories).enter().append('svg')
+      .attr('class', '.flower-box')
+      .attr('height', '300')
+      .attr('width', '300')
+      .attr('viewBox','-150 -150 300 300')
+      .attr('x', (d, i) => i * 200 - 50)
+      .attr('y', d => yStemScale(d.lifespan) - 175)
+
+    const petalLayer = flowerPositionBox.selectAll('.petal-layer')
+      .data((d) => {
+        console.log('data from flower:', d);
+        return [
+          {
+            color: colorsByLanguage[d.languages[2]] || 'none',
+            path: petalPaths[2],
+            petalRotationStart: petalRotationStarts[2],
+            scale: flowerSizeScale(d.lifespan)
+          },
+          {
+            color: colorsByLanguage[d.languages[1]] || 'none',
+            path: petalPaths[1],
+            petalRotationStart: petalRotationStarts[1],
+            scale: flowerSizeScale(d.lifespan)
+          },
+          {
+            color: colorsByLanguage[d.languages[0]] || 'none',
+            path: petalPaths[0],
+            petalRotationStart: petalRotationStarts[0],
+            scale: flowerSizeScale(d.lifespan)
+          },
+        ];
+      })
+      .enter().append('g')
+      .attr('class', 'petal-layer')
+
+    const petal = petalLayer.selectAll('.petal')
+      .data(d => [d,d,d])
+      .enter().append('path')
+      .attr('class', 'petal')
+      .attr('fill', d => d.color)
+      .attr('d', d => d.path)
+      .attr('transform', (d,i) => `rotate(${d.petalRotationStart + i * 120 || 0})scale(${d.scale})`)
 
     const flowerCenter = flowerBed.selectAll('circle')
       .data(repositories).enter().append('circle')
-      .attr('r', 25)
+      .attr('r', 15)
       .attr('cx', (d, i) => 100 + i * 200)
       .attr('cy', d => yStemScale(d.lifespan) - 20)
       .attr('stroke-width', 1)
       .attr('stroke', 'blue')
       .attr('fill', 'blue')
-      //Append the month names to each slice
-
-
-  //   console.log('STEMS', stem)
-  //
-  //   //DRAW ROOTS
-  //   //one root for each branch
-  //   //text path w branch name
-  //   // const root =
-  //   // console.log('ROOTS', roots)
-  }
-  //
+
+      flowerBed.selectAll('.repoName')
+        .data(repositories).enter()
+        .append('text')
+        .append('textPath')
+        .attr('xlink:href', (d, i) => `#myStem${i}`)
+        .text(d => d.name)
+        .attr('font-size', '1.5rem')
+
+      const rootBox = flowerBed.selectAll('.root-box')
+        .data(repositories).enter().append('svg')
+        .attr('class', '.root-box')
+        .attr('height', '200')
+        .attr('width', '200')
+        .attr('viewBox','-150 -150 300 300')
```

Comment:
Why is the viewbox starting at -150, -150?

Reply:
To make the center of the viewbox 0,0 and have the petals still fit without redoing their paths completely. In our original paths the petals were more than 100px long, so it was easiest to make the viewboxes 300x300